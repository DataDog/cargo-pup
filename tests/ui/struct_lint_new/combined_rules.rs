//@compile-flags: --crate-name test_combined_rules
//@compile-flags: --crate-type lib

// This test verifies that multiple struct lint rules can be combined.
// In this case, structs matching 'TestStruct.*' must:
// 1. Match pattern 'TestStruct[A-Z].*' (start with TestStruct followed by a capital letter)
// 2. Not match pattern '.*Forbidden.*' (must not contain the word "Forbidden")

// Violates rule 1 - doesn't have a capital letter after "TestStruct"
pub struct TestStructlowercase { //~ ERROR: Struct must match pattern 'TestStruct[A-Z].*', found 'TestStructlowercase'
    data: i32,
}

// Violates rule 2 - contains "Forbidden"
pub struct TestStructAForbiddenName { //~ ERROR: Struct must not match pattern '.*Forbidden.*'
    name: String,
}

// Satisfies both rules - starts with "TestStruct" + capital letter, and doesn't contain "Forbidden"
pub struct TestStructValidName {
    id: u32,
    value: String,
}

// Doesn't trigger any rules because it doesn't match the initial selector pattern "TestStruct.*"
pub struct OtherStruct {
    field: bool,
} 