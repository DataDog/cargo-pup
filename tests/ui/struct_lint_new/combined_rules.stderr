error: Struct must match pattern 'TestStruct[A-Z].*', found 'TestStructlowercase'
  --> tests/ui/struct_lint_new/combined_rules.rs:12:1
   |
LL | pub struct TestStructlowercase {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Rename this struct to match the pattern 'TestStruct[A-Z].*'
   = note: Applied by cargo-pup rule 'combined_naming_test'.
   = note: `#[deny(struct_lint_must_be_named)]` on by default

error: Struct must not match pattern '.*Forbidden.*'
  --> tests/ui/struct_lint_new/combined_rules.rs:17:1
   |
LL | pub struct TestStructAForbiddenName {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Choose a name that doesn't match this pattern
   = note: Applied by cargo-pup rule 'combined_naming_test'.
   = note: `#[deny(struct_lint_must_not_be_named)]` on by default

error: aborting due to 2 previous errors

